src/lightning/fabric/strategies/fsdp.py

```diff
@@ -451,39 +446,70 @@ def load_checkpoint(
             )
         module_key, module = list(modules.items())[0]
 
-        state_dict_ctx = _get_sharded_state_dict_context(module)
-        reader = FileSystemReader(path=path)
-
-        with state_dict_ctx:
-            module_state = {module_key: module.state_dict()}
-            load_state_dict(module_state, reader)
-            module.load_state_dict(module_state[module_key])
+        if _is_sharded_checkpoint(path):
```

Comment:
Apologies to the reviewers for the bad diff. The previous code is indented and unchanged under this if statement, while the newly added code is under the "else".